### ISSUE TITLE
Your net structure is different from the article decribe. The order of the structure is not correct.

### ISSUE THREAD

**Issue by Yang-HangWA** [j40903272/MalConv-keras#3]
hi,@j40903272,there is pull request for you.Your code is good,but i want to join you to make it better.
The structure of malware is:
![image](https://user-images.githubusercontent.com/38546249/47831783-0400de00-ddcd-11e8-9261-7a2484d51334.png)

Your code struct is as follow:
![image](https://user-images.githubusercontent.com/38546249/47831720-b3898080-ddcc-11e8-8fcb-794935f88f79.png)

The change struct is:
![image](https://user-images.githubusercontent.com/38546249/47831626-44ac2780-ddcc-11e8-81cd-8e84ec980c3e.png)
So,you can see the correct struct is the second one.And i have test both two,the second one also has a better performance.

**Comment by j40903272** [comment]
Hi, @Yang-HangWA , very appreciated for your advice. The current model architecture is according to the picture on NVIDIA website(https://devblogs.nvidia.com/malware-detection-neural-networks/) which is from the original author.

![image](https://user-images.githubusercontent.com/15309569/47866866-e702f400-de3a-11e8-8dc0-e9a57754cd28.png)

I believe that different model architecture would be suitable for different datasets, but let us leave it the same as the original one proposed in the paper. Thanks again.

**Comment by Yang-HangWA** [comment]
Hi,@j40903272 . I am really happy for your reply, but i still have some questions about this model. My dataset is from ordinary daily data collection, the number of our train data is about 70w+ file, and the ratio of malware and harmless file is 1:1. I train  both of the model on this dataset for a few days with what you default setting. The model in your code get a unbalanced  result, it performance better on the harmless file. Actually，i just want to figure out what's wrong with this result. Strangely,the original model get a balanced accuracy rate. And i also change some dataset to verify this strange performance.Also, do you have any better model to solve this problem?

**Comment by Yang-HangWA** [comment]
@j40903272 Looking forward to your reply!

**Comment by j40903272** [comment]
Hi, @Yang-HangWA . It's hard to tell why you've got an unbalanced result with such insufficient information.  There are different architectures such as [Deep Convolutional Malware Classifiers Can Learn from Raw Executables and Labels Only](https://openreview.net/forum?id=HkHrmM1PM). Perhaps you could consider hand crafted features which usually out perform end-to-end deep models.

**Comment by j40903272** [comment]
Hi @Yang-HangWA,

Collecting malware are quite simple, but collecting many latest benign files are very difficult for me. How do you collect them usually?

**Comment by Yang-HangWA** [comment]
@j40903272 I was working in a  company that easy enough to collect benign, there is a large distributed database to store the file. However, I leave the company a month ago and working in a game company now and the data there can't be shared. I think you should find a big security company to collaborate for this job.
